new-lamassu-admin/src/pages/Customers/components/PhotosCard.js

```diff
@@ -0,0 +1,167 @@
+      return ['path', value]
+    }
+    if (key === 'txCustomerPhotoAt' || key === 'frontCameraAt') {
+      return ['date', value]
+    }
+    return pair
+  }
+
+  const addPhotoDir = R.map(it => {
+    const hasFrontCameraData = R.has('id')(it)
+    return hasFrontCameraData
+      ? { ...it, photoDir: 'operator-data/customersphotos' }
+      : { ...it, photoDir: 'front-camera-photo' }
+  })
+
+  const standardizeKeys = R.map(
+    R.compose(R.fromPairs, R.map(mapKeys), R.toPairs)
+  )
+
+  const filterByPhotoAvailable = R.filter(
+    tx => !R.isNil(tx.date) && !R.isNil(tx.path)
+  )
+
+  const photosData = filterByPhotoAvailable(
+    addPhotoDir(standardizeKeys(R.append(frontCameraData, txPhotosData)))
+  )
+
+  const singlePhoto = R.head(photosData)
+
+  const isPhotoRollAvailable = () => {
+    return !singlePhoto
+  }
+
+  return (
+    <>
+      <Paper className={classes.photo} elevation={0}>
+        <ButtonBase
+          disabled={isPhotoRollAvailable()}
+          className={classes.button}
+          onClick={() => {
+            setPhotosDialog(true)
+          }}>
+          {singlePhoto ? (
+            <div className={classes.container}>
+              <img
+                className={classes.img}
+                src={`${URI}/${singlePhoto.photoDir}/${singlePhoto.path}`}
+                alt=""
+              />
+              <circle className={classes.circle}>
+                <div>
+                  <Info2>{photosData.length}</Info2>
+                </div>
+              </circle>
+            </div>
+          ) : (
+            <CrossedCameraIcon />
+          )}
+        </ButtonBase>
+      </Paper>
+      <InformativeDialog
+        open={photosDialog}
+        title={`Photo roll`}
+        data={<PhotosCarousel photosData={photosData} />}
+        onDissmised={() => {
+          setPhotosDialog(false)
+        }}
+      />
+    </>
+  )
+})
+
+export const PhotosCarousel = memo(({ photosData }) => {
+  const classes = useStyles()
+  const [currentIndex, setCurrentIndex] = useState(0)
+
+  const isFaceCustomerPhoto = !R.has('id')(photosData[currentIndex])
+
+  const slidePhoto = index => setCurrentIndex(index)
+
+  return (
+    <>
+      <Carousel
+        navButtonsProps={{
+          style: {
+            backgroundColor: 'transparent',
+            borderRadius: 0,
+            fontSize: 100
+          }
+        }}
+        className={classes.slideButtons}
+        autoPlay={false}
+        indicators={false}
+        navButtonsAlwaysVisible={true}
+        next={activeIndex => slidePhoto(activeIndex)}
+        prev={activeIndex => slidePhoto(activeIndex)}>
+        {photosData.map((item, i) => (
+          <div>
+            <div className={classes.imgWrapper}>
+              <img
+                className={classes.imgInner}
+                src={`${URI}/${item?.photoDir}/${item?.path}`}
+                alt=""
+              />
+            </div>
+          </div>
+        ))}
+      </Carousel>
```

Comment:
Could this be moved into a separate component for general use in other parts of the system (apart from line 109)? Should be a simple, but beneficial refactoring